src/pages/settings/Payments/TransferBalancePage.js

```diff
@@ -1,27 +1,246 @@
+import variables from '../../../styles/variables';
+import ExpensifyText from '../../../components/ExpensifyText';
+import Button from '../../../components/Button';
+import FixedFooter from '../../../components/FixedFooter';
+import CurrentWalletBalance from '../../../components/CurrentWalletBalance';
+import * as paymentPropTypes from './paymentPropTypes';
+import * as PaymentMethods from '../../../libs/actions/PaymentMethods';
+import PaymentUtils from '../../../libs/PaymentUtils';
+import userWalletPropTypes from '../../EnablePayments/userWalletPropTypes';
 
 const propTypes = {
+    /** User's wallet information */
+    userWallet: userWalletPropTypes.userWallet,
+
+    /** Array of bank account objects */
+    bankAccountList: paymentPropTypes.bankAccountListPropTypes,
+
+    /** Array of card objects */
+    cardList: paymentPropTypes.cardListPropTypes,
+
+    /** Wallet balance transfer props */
+    walletTransfer: paymentPropTypes.walletTransferPropTypes,
+
     ...withLocalizePropTypes,
 };
 
-const TransferBalancePage = props => (
-    <ScreenWrapper>
-        <KeyboardAvoidingView>
-            <HeaderWithCloseButton
-                title={props.translate('common.transferBalance')}
-                shouldShowBackButton
-                onBackButtonPress={() => Navigation.goBack()}
-                onCloseButtonPress={() => Navigation.dismissModal(true)}
-            />
-        </KeyboardAvoidingView>
-    </ScreenWrapper>
-);
+const defaultProps = {
+    userWallet: {},
+    bankAccountList: [],
+    cardList: [],
+    walletTransfer: {
+        transferAmount: 0,
+    },
+};
+
+class TransferBalancePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.paymentTypes = [
+            {
+                key: CONST.WALLET.TRANSFER_METHOD_TYPE.INSTANT,
+                title: this.props.translate('transferAmountPage.instant'),
+                description: this.props.translate('transferAmountPage.instantSummary', {
+                    amount: this.props.numberFormat(
+                        0.25,
+                        {style: 'currency', currency: 'USD'},
+                    ),
+                }),
+                icon: Expensicons.Bolt,
+                type: CONST.PAYMENT_METHODS.DEBIT_CARD,
+            },
+            {
+                key: CONST.WALLET.TRANSFER_METHOD_TYPE.ACH,
+                title: this.props.translate('transferAmountPage.ach'),
+                description: this.props.translate('transferAmountPage.achSummary'),
+                icon: Expensicons.Bank,
+                type: CONST.PAYMENT_METHODS.BANK_ACCOUNT,
+            },
+        ];
+
+        this.transferBalance = this.transferBalance.bind(this);
+
+        const selectedAccount = this.getSelectedAccount();
+        PaymentMethods.saveWalletTransferAmountAndAccount(
+            this.props.userWallet.currentBalance,
+            selectedAccount ? selectedAccount.id : '',
+        );
```

Comment:
Does it make sense to do this in the `constructor()`? If the `currentBalance` changes the user will transfer whatever we first had when the page opened and not the new balance.

Reply:
Actually, the balance is transferred to the backend. A just takes thee account Di where the user wants to transfer. But there is no other way on the app where wallet values will update automatically. The wallet is only refreshed when we open some pages and the First-time app loads.

But I agree I can subscribe to wallet wherever I need the transfer amount.

Reply:
I wanted to remove the temporary data that is added for wallet transfer. we can't clear the data(set that to null) before Confirm modal is shown as the same key `ONYXKEYS.WALLET_TRANSFER.shouldShowConfirmModal` would be used to show the modal.

After the modal is shown and use close it. We can remove all the temporary data from Onyx by setting it to null.

@marcaaron

Reply:
Sorry, I don't understand the explanation. Can you try asking this in the form of a question or point to some examples?

Reply:
So we use an Onyx key to store the temporary data used for wallet transfer. And the same key is used to determine the isVisible state of ConfirmModal. 
1. After the transfer is done, we don't need the temporary data and thus we have to set the Onyx key to null to clear it.
2. Confirm Modal is shown to user after API call is made and we need `shouldShowConfirmModal` to open this.
3. If we set the Onyx data to null after API call we won't be able to determine the state for ConfirmModal. 

we need a trigger point to clear the Onyx data after wallet transfer i.e. a place where we can set the Key to null. I think the best place would be after user has confirmed the Transfer.

Reply:
Sorry, not sure if I totally understand still. But can we solve this problem by using a new key for the modal visibility flag to separate it from the wallet data? That way we can reset or clear the data however we want and have separate controls to show/hide the modal?

Reply:
>we have to set the Onyx key to null to clear it

Unsure why we need to set this `null` - what happens if we don't do this?

Reply:
Nothing. It will just remove those keys from storage but I think storage is never a problem for us. So I am correcting it now.